### ISSUE TITLE
C++ for posterior_predict

### ISSUE THREAD

**Issue by rtrangucci** [stan-dev/rstanarm#117]
#### Summary:

`posterior_predict` can likely be sped up by using C++ functions in place of apply functions

#### Description:

Maybe using `expose_stan_functions` on functions defined in the model could simplify speeding up `posterior_predict`

**Comment by bgoodri** [comment]
Needs https://github.com/stan-dev/stan/pull/2071 first

**Comment by jgabry** [comment]
I believe that is now merged

**Comment by bgoodri** [comment]
Yes, although it won't help much until 2.13.x  is released.

**Comment by wds15** [comment]
Hi!

apply functions in R should already give good speed. All the heavy lifting is done on the C side already. While I won't stop anyone from doing this... I am not sure if this buys so much. If it does, that would be interesting.

Sebastian

**Comment by jgabry** [comment]
Yeah I'd be curious to know if it makes a difference. We're already using
sapply so probably not, if you're right. We might also be able to get a
speedup just by switching to vapply but it would be tiny probably.
